lib/pf/Connection/Profile.pm

```diff
@@ -568,10 +568,6 @@ sub findScan {
 
     $node_attributes ||= node_attributes($mac);
     my $os = $node_attributes->{'device_type'};
-    unless(defined $os){
```

Comment:
We need to move this logic in pf::scan::matchOS after the 'return true if @oses == 0' so that it doesn't attempt to match with an undefined OS